### PR TITLE
Add Mikrotik LLDP discovery

### PR DESCRIPTION
**Pull request by erotel**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

RouterOS must be > 6.38

mikrotik-lldp

### CODE REVIEW

**Comment by laf** [comment]
Nice work @erotel :)

out of interest can the other LLDP code further down not be re-used?

**Comment by erotel** [comment]
Standard LLDP can not be used because Mikrotik does not meet the LLDP standard

**Comment by laf** [comment]
> Standard LLDP can not be used because Mikrotik does not meet the LLDP standard

Oh how I laughed :(

File: includes/discovery/discovery-protocols.inc.php [hunk 1]
```diff
@@ -81,7 +81,45 @@
     echo PHP_EOL;
 }//end if
 
-if (($device['os'] == 'pbn' || $device['os'] == 'bdcom') && Config::get('autodiscovery.xdp') === true) {
+if (($device['os'] == 'routeros') && Config::get('autodiscovery.xdp') === true) {
+    echo ' LLDP-MIB: ';
+    $lldp_array  = snmpwalk_group($device, 'lldpRemEntry', 'LLDP-MIB', 3);
+    $lldp_ports = snmpwalk_group($device, 'mtxrInterfaceStatsName', 'MIKROTIK-MIB');
```
Comment (murrant) [inline includes/discovery/discovery-protocols.inc.php]: Do not unconditionally do 3 snmpwalks here. 

If the first one fails, there is no point in doing the other two.

**Comment by murrant** [comment]
Looks great @erotel, looks good to merge after the small change I mentioned

**Comment by erotel** [comment]
@laf The Mikrotik has a lot of things wrong

standart lldp in edge core

```
root@speedtest:/opt/librenms# snmpwalk -v2c -c public -OQUsetX -m LLDP-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/mikrotik udp:192.168.224.224:161 lldpRemTable
lldpRemChassisIdSubtype[0][5][5] = 5
lldpRemChassisIdSubtype[0][5][9] = 4
lldpRemChassisIdSubtype[0][24][4] = 7
lldpRemChassisIdSubtype[0][24][6] = 7
lldpRemChassisIdSubtype[0][27][3] = 4
lldpRemChassisId[0][5][5] = "01 0A 85 54 C6 "
lldpRemChassisId[0][5][9] = "00 1B B1 04 B6 50 "
lldpRemChassisId[0][24][4] = "RAy17_L"
lldpRemChassisId[0][24][6] = "RAy17_U"
lldpRemChassisId[0][27][3] = "CC 37 AB C4 5D 30 "
lldpRemPortIdSubtype[0][5][5] = 3
lldpRemPortIdSubtype[0][5][9] = 5
lldpRemPortIdSubtype[0][24][4] = 7
lldpRemPortIdSubtype[0][24][6] = 7
lldpRemPortIdSubtype[0][27][3] = 3
```
and Mirkotik

```
root@speedtest:/opt/librenms# snmpwalk -v2c -c public -OQUsetX -m LLDP-MIB -M /opt/librenms/mibs:/opt/librenms/mibs/mikrotik udp:10.133.0.8:161 lldpRemTable
lldpRemChassisIdSubtype[1] = 4
lldpRemChassisIdSubtype[2] = 4
lldpRemChassisIdSubtype[3] = 4
lldpRemChassisIdSubtype[4] = 4
lldpRemChassisIdSubtype[5] = 4
lldpRemChassisIdSubtype[6] = 4
lldpRemChassisIdSubtype[7] = 4
lldpRemChassisIdSubtype[8] = 4
lldpRemChassisIdSubtype[10] = 4
lldpRemChassisIdSubtype[11] = 4
lldpRemChassisIdSubtype[12] = 4
lldpRemChassisIdSubtype[13] = 4

```

File: includes/discovery/discovery-protocols.inc.php [hunk 2]
```diff
@@ -81,7 +81,47 @@
...
 }//end if
 
-if (($device['os'] == 'pbn' || $device['os'] == 'bdcom') && Config::get('autodiscovery.xdp') === true) {
+if (($device['os'] == 'routeros') && Config::get('autodiscovery.xdp') === true) {
+    echo ' LLDP-MIB: ';
+    $lldp_array  = snmpwalk_group($device, 'lldpRemEntry', 'LLDP-MIB', 3);
+    if (!empty($lldp_array)) {
+        $lldp_ports = snmpwalk_group($device, 'mtxrInterfaceStatsName', 'MIKROTIK-MIB');
+        $lldp_ports_num = snmpwalk_group($device, 'mtxrNeighborInterfaceID', 'MIKROTIK-MIB');
+    }
```
Comment (laf) [inline includes/discovery/discovery-protocols.inc.php]: May as well move the `foreach ($lldp_array)` into this if check, no point in even attempting to loop through the empty array.

**Pull request merged**